### PR TITLE
Set up tabulation program and macros

### PR DESCRIPTION
**Pull request by maggiesu0725**

### CODE REVIEW

**Comment by maggiesu0725** [comment]
@lhendey I have set up the programs and macros for the tabulation. Details in the assign county macro may change and there's two county variable got assigned, one is for merging with income limits, the other is for tabulation, they may be or not be the same so I have two there. The difference is also reflected in the actual tabulation programs.

For the hud_inc program, the spreasheet doesn't have 120% and 200% so I calcuated them based on the 80% one. I also noticed for at least one year, one category is not available so have to modify the spreadsheet or the macro to reflect that.

Please take a look and let me know if there's any major steps I'm missing. Otherwise I will work on refining the results. I'm getting sick and going home for the day now.

**Comment by lhendey** [comment]
on householder populatino for projections.sas  - this has an error for me and isn't running. 

-  you don't have the NCHsg library on it. Remove RegHsg

- remove county format. 
- can you look to see if there are any GQ=5 ? (currently commented out) They aren't really households. (the ones we removed in DC for example - had 20 people in one house...) 
-i haven't reviewed closely but maybe worth creating one var that you'd collapse on that's County or PUMA depending on what the collapse is on?

**Comment by maggiesu0725** [comment]
@lhendey could you take a look at the assign_NCcounty macro?  I think the part where I'm assigning multiple PUMA to one county fips is not working properly but don't know what's the correct grammar to do it.

File: Prog/Tabulate householder population for projection.sas [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+	run;
+
+	data Household_&year._2 ;
+	merge Household_&year. Inc_&year.;
+	by county_char ;
+	run;
+
+	data Householddetail_&year.;
+		set Household_&year._2;
+		keep race hispan age hhincome hhincome_a pernum relate gq county hhwt perwt year serial numprec race1 agegroup hud_inc totpop_&year. l50_1- l50_8 l80_1- l80_8  median&year.   ;
```
Comment (lhendey) [inline Prog/Tabulate householder population for projection.sas]: hhincome_a is not initialized.  remove.

File: Prog/Tabulate householder population for projection.sas [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+		 %Hud_inc_NCHsg( hhinc=hhincome_a, hhsize=numprec )
```
Comment (lhendey) [inline Prog/Tabulate householder population for projection.sas]: do we actually need the income in this program at all? i'd remove all pieces that aren't necessary to this program. (also you'd want to adjust for inflation..)

File: Prog/Tabulate householder population for projection.sas [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+	data Household_&year. ;
+		set Ipums.Acs_&year._NC;
+
+		%assign_NCcounty;
+		%assign_NCcounty;
+		county_char = put(county, 5.);
+	run;
+
+	data Inc_&year. ;
+	set NCHsg.IncomeLimits_&year. (where= (State=37));
```
Comment (lhendey) [inline Prog/Tabulate householder population for projection.sas]: see comment below -if we don't need to merge on income limits in the this program lets not do it.

File: Prog/Tabulate householder population for projection.sas [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+
+		totpop_&year. = 1;
+		run;
+
+		proc freq data=Householddetail_&year.;
+		  tables race1 * agegroup  / list missing;
+		run;
+
+		proc sort data=Householddetail_&year.;
+		by county2 agegroup race1 relate;
```
Comment (lhendey) [inline Prog/Tabulate householder population for projection.sas]: county2 not found.

File: Prog/Tabulate householder population for projection.sas [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+data Householddetail_2013r;
+	set Householddetail_2013;
+
+if serial in (560493, 255313) then relate=12;
+
+run; 
+*/
+
+data fiveyeartotal;
+set Householddetail_2013_3 Householddetail_2014_3 Householddetail_2015_3 Householddetail_2016_3 Householddetail_2017_3;
```
Comment (lhendey) [inline Prog/Tabulate householder population for projection.sas]: none of these datasets exist.

File: Prog/Tabulate householder population for projection.sas [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+%mend householdinfo;
+
+%householdinfo(2013);
+%householdinfo(2014);
+%householdinfo(2015);
+%householdinfo(2016);
+%householdinfo(2017);
+
+
+/*change serial 560493 and 255313 in 2013 from HoH to other nonrelative as they really reflect GQ (GQ=5) and those households have 11 and 20 people (not related)*/
```
Comment (lhendey) [inline Prog/Tabulate householder population for projection.sas]: you need this step.  there are 7 households with gq=5 that really are group homes.  please reassign to other nonrelative.

File: Prog/Tabulate householder population for projection.sas [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+
+proc export data = distribution_3
+   outfile="&_dcdata_default_path\NCHsg\Prog\Householderratio_NC_&date..csv"
+   dbms=csv
+   replace;
+run;
+
+/*by county*/
+
+proc sort data=fiveyeartotal;
```
Comment (lhendey) [inline Prog/Tabulate householder population for projection.sas]: the rest of this i can't test because there is no county2

**Comment by maggiesu0725** [comment]
@lhendey thanks, I updated the county assignment macros and the two tabulation programs. Please review. Two flags:

For the counties for merging with income limits, I can't think of a better way to assign counties based on the rules we agreed on for each year so I only assigned counties based on the median rents in 2013. 

For county2 macro, I double checked the macro and it looked like I covered all PUMA but if you run code:

> /*make sure all PUMA got assigned a geography for tabulation*/
proc freq data= Household_2013 (where=(county2_char=""));
tables upuma;
run;

There's still missing county2 values that correspond to 4 PUMAs, but I made sure those were assigned in the macro so I'm confused. 

Thanks~

**Comment by maggiesu0725** [comment]
@lhendey The county2 macro is only assign a county2 crosswalk, so after the merge we will actually have the income denominated in each year's currency and the income limit denominated in each year's currency as well. So I don't think we actually need to inflation adjust any income numbers if they were only used to assign each household to a income category? And I will double check the county2 macro and make sure they were based on 2017 median incomes, but I'm pretty sure the ranking of median income are almost identical to 2013.

**Comment by maggiesu0725** [comment]
@lhendey Tabulate householder income for projection.sas ready for review. The assign county macro seems not working when I had both upuma and county in there so had to split into two.

File: Prog/Tabulate householder income for projection.sas [hunk 8]
```diff
@@ -66,63 +68,76 @@ proc format;
...
-        %Hud_inc_NCHsg( hhinc=hhincome_a, hhsize=numprec )
-		  label
-		  hud_inc = 'HUD income category for household'; 
-
 		if hispan=0 then do;
 
 		 if race=1 then race1=1;
 		 else if race=2 then race1=2;
-		 else race1=4;
+         else if race in (4 5 6) then race1=4;
```
[the remark this answers is not in the document]
Comment (maggiesu0725) [reply Prog/Tabulate householder income for projection.sas]: And race/eth categories for…

Steven want the tabulation by:

Nonhispanic white
Nonhispanic black
Nonhispanic Asian and Pacific Islander
Nonhispanic Other (AIAN + 2 or more + other responses)
Hispanic any race

[the remark this answers is not in the document]
Comment (lhendey) [reply Prog/Tabulate householder income for projection.sas]: okay - but both programs still need to be consistent

**Comment by maggiesu0725** [comment]
@lhendey  I updated the programs and they are no longer referencing to the assign_NCcounty macro, which was originally created to merge on the HUD income limits. assign_NCcounty2 and assign_NCcounty3 are supposed to be working together to create the geography entities for the projection (54 total), the only reason I'm breaking them up is because I used two different variables for assigning the geography so they won't work together in one macro.

**Comment by maggiesu0725** [comment]
@lhendey ready for final review.

**Comment by lhendey** [comment]
@maggiesu0725 still have errors in the household population program when i batch submitted.

**Comment by maggiesu0725** [comment]
@lhendey the program works fine in interactive mode, were you batch submitting on your D drive? I want to check the error you got from batch submit but I'm not sure where I should do it-- we usually sync with L: first and batch submit on L, could you do it on D:?

**Comment by maggiesu0725** [comment]
Thanks, I addressed the minor comments and both program should batch submit without error. Sorry for the confusion, I thought we only batch submit on L drive. @lhendey

**Comment by lhendey** [comment]
@maggiesu0725 shouldn't the head_householder column on the population file match the denom on the income file?

**Comment by lhendey** [comment]
@maggiesu0725 yes batch submit on both D & L  to get the clean run.

**Comment by maggiesu0725** [comment]
yes, we should weight by AFFACT2. @lhendey  they are very close but still slightly different by the decimals, I couldn't figure out why.

**Comment by lhendey** [comment]
@maggiesu0725 - can you confirm that the difference is just the use of the perwt vs hhwt?

**Comment by maggiesu0725** [comment]
Yes when using the same weight the numbers are consistent, but we should keep the different weight since one is population and the other is household level right? @lhendey

**Comment by maggiesu0725** [comment]
programs and projections are finished @lhendey

**Pull request merged**